kotest-tests/kotest-tests-spec-parallelism/src/jvmTest/kotlin/com/sksamuel/kotest/parallelism/ProjectConfig.kt

```diff
@@ -52,8 +55,8 @@ object ProjectConfig : AbstractProjectConfig() {
    /** Mark the start of the entire test case. */
    internal val startOfTest: TimeMark = TimeSource.Monotonic.markNow()
 
-   /** Marks when [beforeProject] is called, which should be before any tests are launched. */
-   private var beforeProjectCalled: Duration? = null
+   /** Marks when [beforeProject] is called, which should only be once, and before any tests are launched. */
+   private val beforeProjectCalled = MutableStateFlow(listOf<Duration>())
```

Comment:
what would happen if it's called more than once, so the folling assumption has failed: "should only be once, and before any tests are launched."

Reply:
If it is called more than once then the assertion on line 97 will fail.